### PR TITLE
Fill out ParallelConfig based on docs

### PR DESCRIPTION
**Pull request by RickDT**
There were a few more options we should have in the ParallelConfig type, based on our docs.

I also took out the "Partnership LLC" and "Partnership LP" business type strings that expire this week, h/t @nonrational

### CODE REVIEW

**Review by nonrational — APPROVED** [review]

File: packages/vanilla/src/index.ts [hunk 1]
```diff
@@ -76,7 +76,13 @@ export const loadParallel = (config: ParallelConfig): Promise<Parallel | null> =
...
       // Resolve the initialized Parallel object upon init
       const onInit = () => resolve(Parallel)
-      Parallel.init({ ...config, on_init: onInit })
+      // The `raw_config` option lets you pass in extra values that aren't described in the ParallelConfig type
+      // such as upcoming features that are being tested.
+      const baseConfig = config.raw_config ?? {}
+      const rawlessConfig = { ...config }
+      delete rawlessConfig.raw_config
+      const parallelConfig = { ...baseConfig, ...rawlessConfig }
+      Parallel.init({ ...parallelConfig, on_init: onInit })
```
Comment (bmuller) [inline packages/vanilla/src/index.ts:85]: There are a few additional params that aren't currently in the ParallelConfig - 
 * expected_entity_id
 * expected_entity_type
 * expected_business_name
 * partner_supporting

Comment (RickDT) [reply packages/vanilla/src/index.ts:85]: Ah! I see those on the [SDK Functions](https://developer.parallelmarkets.com/docs/javascript/sdk) page, but not the [SDK Config](https://developer.parallelmarkets.com/docs/javascript/configuration/) page. Should I add them there @bmuller?

Comment (RickDT) [reply packages/vanilla/src/index.ts:85]: Actually, I think those are all options that you can send to `login()`, not the SDK config. I don't have any types in here for `login()`, but that'll be quick. I'll tag you on a follow-up, Brian.

File: packages/vanilla/src/types.ts [hunk 2]
```diff
@@ -31,19 +51,45 @@ export type ParallelApiSuccessCallback = (response: ParallelApiRecord) => void
...
+  last_name?: string
+  expected_entity_id?: string
+  expected_entity_type?: 'self' | BusinessType
+}
 
 export interface Parallel {
   init(options: ParallelConfig): void
   _config: ParallelConfig
-  login: () => void
+  login: (options?: LoginOptions) => void
```
Comment (RickDT) [inline packages/vanilla/src/types.ts:85]: @bmuller I think this is right. Please double-check me 🙏

File: packages/vanilla/src/index.ts (hunk 1 above)
Comment (bmuller) [reply packages/vanilla/src/index.ts:86]: OK - sorry, think I got the above wrong and will try to clarify here.  I was thinking the ParallelConfig was shared w/ the login options, which I don't think is the case now that I look deeper.

There are a number of params that `login()` can take that will override what was used in the initial `init()` call - but not all and `login` shouldn't accept the same type as `init` - so I think what you had initially is correct.  

The ability for `login()` to take in params like `client_id` that override whatever was initially passed to `init` is an undocumented feature that a few of our partners with unique setups use (I think it should stay undocumented, cause it's only used in very unique / rare cases)

File: packages/vanilla/src/types.ts [hunk 3]
```diff
@@ -31,19 +51,45 @@ export type ParallelApiSuccessCallback = (response: ParallelApiRecord) => void
...
+  }
+}
+type SubscriptionHandler = (response: SubscriptionEvent) => void
+
+export interface LoginOptions {
+  email?: string
+  first_name?: string
+  last_name?: string
+  expected_entity_id?: string
+  expected_entity_type?: 'self' | BusinessType
```
Comment (bmuller) [inline packages/vanilla/src/types.ts:79]: Think we need two more here:
```suggestion
  expected_entity_type?: 'self' | BusinessType
  expected_business_name?: string
  partner_supporting?: string
```

File: packages/vanilla/src/types.ts [hunk 4]
```diff
@@ -14,12 +16,30 @@ export type AuthCallbackResult = {
...
-  flow_type?: 'overlay' | 'embed' | 'redirect'
-  scopes?: Array<string>
+  force_accreditation_check?: boolean
+  force_identity_check?: boolean
+  scopes?: Array<ParallelScope>
+  show_dismiss_button?: boolean
+  identity_claim_override_id?: number
+  verbose?: boolean
+  expected_entity_id?: string
+  expected_entity_type?: string
```
Comment (bmuller) [inline packages/vanilla/src/types.ts:40]: Sorry for (my) confusion here - I think we can remove these

```suggestion
```

File: packages/vanilla/src/types.ts (hunk 2 above)
Comment (bmuller) [reply packages/vanilla/src/types.ts:85]: Had to suggestions above - then I think it'll be ✅

File: packages/vanilla/src/index.ts (hunk 1 above)
Comment (RickDT) [reply packages/vanilla/src/index.ts:86]: Cool. I set the options on `login()` as optional altogether and individually. That should cover all bases. In typical use cases, you'll never pass options to login.

**Review by bmuller — APPROVED** [review]

**Pull request merged**